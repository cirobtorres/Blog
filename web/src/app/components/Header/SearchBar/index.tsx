"use client";

import { Search } from "lucide-react";
import { useDebouncedCallback } from "use-debounce";
import { ReadonlyURLSearchParams, useSearchParams } from "next/navigation";
import { useState } from "react";
import slugify from "../../../utils/slugify";
import { getArticles } from "../../../lib/articles";
import Link from "next/link";
import Image from "next/image";
import {
  Dialog,
  DialogClose,
  DialogContent,
  DialogDescription,
  DialogHeader,
  DialogTitle,
  DialogTrigger,
} from "../../Shadcnui/dialog";

type ArticleList = {
  documentId: string;
  title: string;
  slug: string;
  cover: {
    documentId: string;
    url: string;
    alternativeText: string;
  };
}[];

const SearchBar = () => {
  const searchParams = useSearchParams();
  const [sortedList, setSortedList] = useState<ArticleList | null>(null);

  const handleSearch = useDebouncedCallback(async (queryText: string) => {
    const params = new URLSearchParams(searchParams);

    // Delete user's input if it is empty
    if (queryText) {
      params.set("query", queryText);
    } else {
      params.delete("query");
    }

    const sortBySlugfied = slugify(params.get("query") || "");
    if (sortBySlugfied !== "") {
      const { data: articleList } = await getArticles(
        "createdAt:desc",
        sortBySlugfied
      );

      setSortedList(articleList.length === 0 ? [] : articleList.slice(0, 7));
    } else {
      setSortedList(null);
    }
  }, 500);

  return (
    <Dialog>
      <DialogTrigger
        className="w-full rounded-3xl flex-1" // outline-none
      >
        <Trigger />
      </DialogTrigger>
      <DialogContent>
        <DialogHeader>
          <DialogTitle className="sr-only">Buscador de artigos</DialogTitle>
        </DialogHeader>
        <form action="/artigos" className="overflow-x-hidden">
          <InputQuery handleSearch={handleSearch} searchParams={searchParams} />
          <ResultQuery sortedList={sortedList} />
        </form>
      </DialogContent>
    </Dialog>
  );
};

const Trigger = () => {
  return (
    <div className="relative w-full min-h-10 flex items-center rounded-3xl border border-blog-border bg-blog-background-1 transition-all duration-500 hover:bg-blog-background-2 group">
      <Search className="absolute left-3 top-1/2 -translate-y-1/2 size-4 transition-all duration-500 text-blog-foreground-readable group-hover:text-blog-foreground-readable-hover" />
      <p className="pl-12 w-full text-sm text-start transition-all duration-500 group-hover:text-blog-foreground-readable-hover">
        Buscar artigos...
      </p>
    </div>
  );
};

const InputQuery = ({
  searchParams,
  handleSearch,
}: {
  searchParams: ReadonlyURLSearchParams;
  handleSearch: (queryString: string) => void;
}) => {
  return (
    <div className="relative min-h-10 w-full py-4">
      <div className="flex flex-col justify-center">
        <button
          type="submit"
          className="absolute left-4 top-1/2 -translate-y-1/2"
        >
          <Search className="size-4 transition-all duration-500 text-blog-foreground-readable hover:text-blog-foreground-readable-hover" />
        </button>
        <input
          id="query"
          type="query"
          name="query"
          autoComplete="off"
          placeholder="Pesquisar artigos..."
          onChange={(event) => {
            handleSearch(event.target.value);
          }}
          defaultValue={searchParams.get("query")?.toString()}
          className="text-sm w-full border-transparent border-none border-0 outline-none outline-0 pl-12 pr-16 bg-transparent"
        />
      </div>
      <DialogClose className="absolute right-4 top-1/2 -translate-y-1/2 px-2 py-0.5 text-xs font-extrabold rounded border border-blog-border transition-all duration-500 bg-blog-background-2 hover:text-blog-foreground-readable-hover hover:bg-blog-background-1">
        <DialogDescription className="sr-only">
          Fechar buscador
        </DialogDescription>
        ESC
      </DialogClose>
    </div>
  );
};

const ResultQuery = ({ sortedList }: { sortedList: ArticleList | null }) => {
  return (
    <>
      {sortedList && sortedList.length > 0 && (
        <>
          <hr />
          <ul className="flex flex-col py-4">
            {sortedList.map((article) => (
              <li key={article.documentId}>
                <Link
                  href={`/artigos/[documentId]/[slug]`}
                  as={`/artigos/${article.documentId}/${article.slug}`}
                  className="flex items-center gap-3 py-1.5 px-5"
                >
                  <div className="shrink-0 relative size-8">
                    <Image
                      src={`http://127.0.0.1:1337${article.cover.url}`}
                      alt={article.cover.alternativeText}
                      fill
                      sizes="(min-width: 16px) 100vw"
                      className="absolute object-cover"
                    />
                  </div>
                  <p className="text-sm truncate text-blog-foreground-readable">
                    {article.title}
                  </p>
                </Link>
              </li>
            ))}
          </ul>
        </>
      )}
      {sortedList && sortedList.length === 0 && (
        <>
          <hr />
          <div className="py-4">
            <span className="sr-only">Nenhum artigo encontrado</span>
            <p className="text-center text-sm text-blog-foreground-readable">
              Nenhum artigo encontrado
            </p>
          </div>
        </>
      )}
    </>
  );
};

export default SearchBar;
