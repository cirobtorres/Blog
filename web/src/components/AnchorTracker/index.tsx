"use client";

import Link from "next/link";
import { useEffect, useState } from "react";
import {
  Accordion,
  AccordionContent,
  AccordionItem,
  AccordionTrigger,
} from "../shadcnui/accordion";
import { extractAnchors } from "../../utils/anchors";

const AnchorTracker = ({ documentId }: { documentId: string }) => {
  const [anchorList, setAnchorList] = useState<{ [key: string]: string }[]>([]);

  const linkAnchorsListener = () => {
    const sections: NodeListOf<HTMLHeadingElement> | undefined = document
      .getElementById(documentId)
      ?.querySelectorAll("h1, h2, h3, h4, h5, h6");

    let currentSectionIndex = 0;

    if (sections) {
      sections.forEach((section, index) => {
        // Collects offsetTop of each header, if exists
        const sectionTop = section.offsetTop;

        // headerHeight = measurement from blog-content to the top of the page
        // 80 pixels top padding
        // 80 pixels bottom padding
        // 80 pixels bottom margin
        // 136 pixels blog-content-hero

        // Minus 1 because we want the anchor color to update even when we click on it
        // An anchor-link redirects to the exact section top
        // With offsetTop lower by 1 pixel we make sure the section is actually lower than it really is
        // With that the link redirects to within the actual section, not to the top of it
        const headerHeight = 48 + 400 + 480 + 80 - 1;

        // Update currentSectionIndex when changing sections
        // Every index is a heading of any type
        if (window.scrollY >= sectionTop + headerHeight) {
          currentSectionIndex = index;
        }
      });

      const links = document
        .getElementById("link-anchor-tracker")
        ?.querySelectorAll("li a");

      links?.forEach((link, index) => {
        if (index === currentSectionIndex) {
          link.setAttribute("aria-current", "page");
        } else {
          link.setAttribute("aria-current", "false");
        }
      });
    }
  };

  const generatePaddingForSessions = (text: { [x: string]: string }) => {
    return Object.values(text)[0].match(/<h[1][^>]*>(.*?)<\/h[1]>/gi)
      ? "pl-0"
      : Object.values(text)[0].match(/<h[2][^>]*>(.*?)<\/h[2]>/gi)
      ? "pl-3"
      : Object.values(text)[0].match(/<h[3][^>]*>(.*?)<\/h[3]>/gi)
      ? "pl-6"
      : Object.values(text)[0].match(/<h[4][^>]*>(.*?)<\/h[4]>/gi)
      ? "pl-9"
      : Object.values(text)[0].match(/<h[5][^>]*>(.*?)<\/h[5]>/gi)
      ? "pl-12"
      : "pl-15";
  };

  useEffect(() => {
    window.addEventListener("scroll", linkAnchorsListener);
    // cleanup function
    return () => {
      window.removeEventListener("scroll", linkAnchorsListener);
    };
  }, []);

  useEffect(() => {
    const contentElement = document.getElementById(documentId);
    if (contentElement) {
      const anchors = extractAnchors(contentElement.innerHTML);
      setAnchorList(anchors);
    }
  }, [documentId]);

  return (
    <nav>
      <Accordion
        type="single"
        collapsible
        defaultValue="item-1"
        className={
          "w-full" +
          " self-start max-w-72 sticky top-12" +
          " max-[800px]:self-auto max-[800px]:max-w-full max-[800px]:static max-[800px]:pt-0"
        }
      >
        <AccordionItem value="item-1">
          <AccordionTrigger>Conte√∫do</AccordionTrigger>
          <AccordionContent>
            {anchorList?.map((text, index) => (
              <li key={index} className="mb-1">
                <Link
                  href={`#${Object.keys(text)}`}
                  aria-current={index === 0 ? "page" : "false"} // When pages load, the first anchor is supposed to be the colored one
                  className={
                    `flex text-sm transition-colors duration-500 break-words aria-current:text-blog-foreground-highlight` +
                    ` aria-current:hover:text-blog-foreground-readable-hover hover:text-blog-foreground-readable-hover ${generatePaddingForSessions(
                      text
                    )}`
                  }
                >
                  {
                    Object.values(text)[0].replace(/<\/?h[1-6][^>]*>/gi, "")
                    // Replaces <h2>Example Title</h2> to Example Title
                  }
                </Link>
              </li>
            ))}
          </AccordionContent>
        </AccordionItem>
      </Accordion>
    </nav>
  );
};

export default AnchorTracker;
