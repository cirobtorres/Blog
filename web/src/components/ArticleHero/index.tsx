"use server";

import Image from "next/image";
import { formatDateToCustomFormat } from "../../functions/dates";
import BreadCrumb from "../BreadCrumb";
import { getAuthor } from "../../lib/author";
import { Skeleton } from "../shadcnui/skeleton";
import Link from "next/link";
import { Suspense } from "react";

export default async function ArticleHero(article: Article) {
  return (
    <>
      <Hero article={article} />
      <ArticleImage cover={article.cover} />
    </>
  );
}

const Hero = ({ article }: { article: Article }) => {
  return (
    <div className="min-h-[30rem] py-8 mb-4 flex items-center bg-blog-background-2">
      <div className="h-full grid grid-cols-article max-lg:grid-cols-article-800 mx-auto items-center max-w-screen-2xl">
        <div className="col-start-2 max-lg:col-start-1 ml-8 mr-4 max-lg:ml-4">
          <BreadCrumb title={article.title} />
          <div className="blog-center-content flex flex-col gap-4">
            <h1 className="text-5xl leading-[4rem] font-extrabold break-words">
              {article.title}
            </h1>
            <p className="text-2xl break-words text-[hsl(0,0%,55%)]">
              {article.description}
            </p>
            <div className="flex gap-8">
              <Suspense
                fallback={
                  <div className="flex gap-4 items-center">
                    <Skeleton className="w-10 h-10 shrink-0 overflow-hidden rounded-full" />
                    <Skeleton className="w-24 h-4 rounded-full" />
                  </div>
                }
              >
                <Author documentId={article.author.documentId} />
              </Suspense>
              <div className="flex flex-col justify-center">
                <p className="text-base">
                  <small>
                    Criado:{" "}
                    <time>{formatDateToCustomFormat(article.createdAt)}</time>
                  </small>
                </p>
                {article.updatedAt && (
                  <p className="text-base leading-3">
                    <small>
                      Atualizado:{" "}
                      <time>{formatDateToCustomFormat(article.updatedAt)}</time>
                    </small>
                  </p>
                )}
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

const Author = async ({ documentId }: { documentId: string }) => {
  const { data: author }: { data: Author } = await getAuthor(documentId);
  return (
    <div className="flex gap-4 items-center">
      <Link href="/sobre-mim" className="group">
        <div className="relative flex w-10 h-10 shrink-0 overflow-hidden rounded-full">
          <Image
            src={`http://127.0.0.1:1337${author.avatar.url}`}
            alt={author.avatar.alternativeText}
            fill
            className="transition-all duration-500 absolute object-cover group-hover:brightness-50"
          />
        </div>
      </Link>
      <Link href="/sobre-mim">
        <p className="transition-colors duration-500 hover:text-blog-foreground-readable-hover text-blog-foreground-highlight">
          <strong>{author.name}</strong>
        </p>
      </Link>
    </div>
  );
};

const ArticleImage = ({ cover }: { cover: Cover }) => {
  return (
    <div
      className={
        "grid grid-cols-article" +
        " max-lg:grid-cols-article-800" +
        " max-w-screen-2xl mx-auto mb-20"
      }
    >
      <figure
        className={
          "flex flex-col gap-3 h-[25rem]" +
          " col-start-2 max-lg:col-start-1" +
          " ml-4 mr-4 pl-4 max-lg:pl-0"
        }
      >
        <div className="shrink-0 relative h-full">
          <Image
            src={`http://127.0.0.1:1337${cover.url}`}
            alt={cover.alternativeText}
            fill
            className="absolute object-cover"
          />
        </div>
        <figcaption className="text-xs">{cover.caption}</figcaption>
      </figure>
    </div>
  );
};
