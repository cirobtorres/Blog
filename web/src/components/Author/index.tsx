import Link from "next/link";
import Image from "next/image";
import { Skeleton } from "../Shadcnui/skeleton";

const Author = ({ author }: { author: Author }) => {
  return (
    <div data-id="author" className="flex gap-2 items-center">
      <Link href="/sobre" className="group">
        <div className="relative flex size-10 shrink-0 overflow-hidden rounded-full">
          <Image
            src={
              author.avatar
                ? `http://127.0.0.1:1337${author.avatar.url}`
                : "/images/not-authenticated.png"
            }
            alt={
              author.avatar
                ? author.avatar.alternativeText
                : `Avatar de ${author.name}`
            }
            fill
            sizes="(max-width: 40px) 100vw"
            className="transition-all duration-500 absolute object-cover group-hover:brightness-50"
          />
        </div>
      </Link>
      <Link href="/sobre">
        <p className="transition-colors duration-500 hover:text-blog-foreground-readable-hover text-blog-foreground-highlight">
          <strong>@{author.name}</strong>
        </p>
      </Link>
    </div>
  );
};

const AuthorSkeleton = () => {
  return (
    <div className="flex gap-4 items-center">
      <Skeleton className="size-10 shrink-0 rounded-full" />
      <Skeleton className="w-20 h-4 shrink-0 rounded-full" />
    </div>
  );
};

export default Author;
export { AuthorSkeleton };
