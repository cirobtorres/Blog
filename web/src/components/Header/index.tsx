"use client";

import Link from "next/link";
import { useEffect, useRef } from "react";
import DarkModeToggle from "../DarkModeToggle";
import ProgressBar from "./ProgressBar";

const FloatingHeader = ({ documentId }: { documentId: string }) => {
  const headerRef = useRef<HTMLElement>(null);

  const hideNavbarListener = () => {
    let prevScrollPos = window.scrollY;
    const threshold = 400 + 480 + 80; // threshold to maintain header on top, beyond which it is allowed to hide

    const handleScroll = () => {
      const currScrollPos = window.scrollY;
      if (currScrollPos < threshold || currScrollPos < prevScrollPos) {
        if (headerRef.current) headerRef.current.style.top = "0";
      } else {
        if (headerRef.current) headerRef.current.style.top = "-48px"; // 48px = header height
      }
      prevScrollPos = currScrollPos;
    };
    return handleScroll;
  };

  useEffect(() => {
    const handleScroll = hideNavbarListener();
    window.addEventListener("scroll", handleScroll);
    // Cleanup function
    return () => {
      window.removeEventListener("scroll", handleScroll);
    };
  }, []);

  return (
    <header
      ref={headerRef}
      id="floating-header"
      className="fixed h-12 w-full backdrop-blur-sm shrink-0 [z-index:10] top-0 transition-[top] duration-300 bg-blog-background-backdrop"
    >
      <HeaderContent />
      <ProgressBar documentId={documentId} />
    </header>
  );
};

const StaticHeader = () => {
  return (
    <header
      id="floating-header"
      className="h-12 w-full backdrop-blur-sm shrink-0 bg-blog-background-backdrop"
    >
      <HeaderContent />
    </header>
  );
};

const HeaderContent = () => {
  return (
    <div className="h-full max-w-screen-2xl px-4 mx-auto">
      <nav className="w-full h-full flex items-center justify-between">
        <div className="h-full flex items-center justify-center gap-4">
          <div className="flex items-center justify-between">
            <Link
              href="/"
              className="flex items-center justify-between text-sm uppercase font-extrabold h-full transition-colors duration-500 text-blog-foreground-readable hover:text-blog-foreground-readable-hover"
            >
              HOME
            </Link>
          </div>
          <ul
            className={
              "flex gap-4 items-center justify-between" +
              " [&_li]:flex [&_li]:items-center [&_li]:justify-between" +
              " [&_li_a]:transition-colors [&_li_a]:duration-500 [&_li_a]:text-sm" +
              " [&_li_a]:text-blog-foreground-readable [&_li_a]:font-[500] hover:[&_li_a]:text-blog-foreground-readable-hover"
            }
          >
            <li>
              <Link href="/sobre-mim">Sobre mim</Link>
            </li>
          </ul>
        </div>
        <ul className="h-full flex items-center justify-between gap-2">
          <li className="flex justify-center items-center h-full">
            <DarkModeToggle />
          </li>
        </ul>
      </nav>
    </div>
  );
};

export { FloatingHeader, StaticHeader };
