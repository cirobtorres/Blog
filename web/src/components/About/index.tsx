import { convertToHTML } from "@/utils/convertToHtml";
import { ParseRichTextBlocks } from "../Article/ArticleContent/ParseBlocks";

export default async function AboutContent({ about }: { about: About }) {
  return about.blocks.map(async (block) => {
    const richTextToRenderA = await convertToHTML(block.body);
    return (
      <ParseRichTextBlocks
        key={`shared.rich-text-${block.id}`}
        body={richTextToRenderA}
      />
    );
  });
}
