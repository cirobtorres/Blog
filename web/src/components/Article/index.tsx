"use server";

import AnchorTracker from "./AnchorTracker";
import { BackToTopButton } from "./BackToTopButton";
import ArticleContent from "./ArticleContent";

const Article = ({ documentId, content }: ArticleContent) => {
  return (
    <section className="relative pb-12 px-4">
      <div className="max-w-screen-2xl mx-auto grid gap-4 grid-cols-article max-lg:grid-cols-article-1024 max-[800px]:grid-cols-article-800">
        <AnchorTracker documentId={documentId} />
        <ArticleContent documentId={documentId} content={content} />
        <div className="max-lg:hidden mx-auto col-start-3 max-lg:col-start-auto">
          <BackToTopButton documentId={documentId} />
        </div>
      </div>
    </section>
  );
};

export default Article;
