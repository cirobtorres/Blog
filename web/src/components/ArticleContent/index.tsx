"use server";

import AnchorTracker from "../AnchorTracker";
import { BackToTopButton } from "../BackToTopButton";
import Article from "./Article";

// Format article content to highlight code blocks with Highlight.js and creates anchor ids for page navigation
const ArticleContent = ({
  id,
  content,
}: {
  id: string;
  content: { __component: string; id: number; title?: string; body?: string }[];
}) => {
  return (
    <div
      className={
        "relative max-w-screen-2xl mx-auto pb-12 px-4" +
        " grid gap-4 grid-cols-article max-lg:grid-cols-article-1024 max-[800px]:grid-cols-article-800"
      }
    >
      <AnchorTracker contentId={id} />
      <Article id={id} content={content} />
      <div className="max-lg:hidden">
        <BackToTopButton contentId={id} />
      </div>
    </div>
  );
};

export default ArticleContent;
