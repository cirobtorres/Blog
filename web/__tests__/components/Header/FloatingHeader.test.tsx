import { act, render, screen, waitFor } from "@testing-library/react";
import { FloatingHeader } from "../../../src/components/Header";

jest.mock("next/navigation", () => ({
  // HeaderContent contains multiple nested components.
  // BlogNavigationMenu uses the usePathname hook, so we need to mock it.
  // SearchBar uses the useSearchParams hook, so we need to mock it as well.
  usePathname: jest.fn(() => "/"),
  useSearchParams: jest.fn(() => new URLSearchParams()),
}));

jest.mock("../../../src/service/about", () => ({
  getAbout: jest.fn().mockResolvedValue({
    data: {
      documentId: "12345",
      title: "My Blog",
      github_link: "https://github.com/johndoe",
      github_blog_link: "https://github.com/johndoe/blog",
      createdAt: "2022-01-01",
      updatedAt: "2022-01-02",
      publishedAt: "2022-01-01",
      blocks: [],
    },
  }),
}));

describe("FloatingHeader", () => {
  const mockAuthUser = {
    ok: true,
    data: {
      id: 1,
      documentId: "Absh1-19AK3-Po24S",
      username: "johndoe",
      email: "johndoe@gmail.com",
      provider: "google",
      confirmed: true,
      blocked: false,
      createdAt: "2025-03-10T23:11:17.381Z",
      updatedAt: "2025-03-10T23:11:17.381Z",
      publishedAt: "2025-03-10T23:11:17.381Z",
    },
    error: null,
  };

  const mockUnauthUser = {
    ok: false,
    data: null,
    error: {
      status: 404,
      name: "",
      message: "Not Found",
    },
  };

  beforeAll(() => {
    window.scrollTo = jest.fn();
  });

  it("renders the authenticated (user) floating header component", async () => {
    await act(async () => {
      render(<FloatingHeader currentUser={mockAuthUser} />);
    });
    const floatingHeader = screen.getByTestId("floating-header");
    expect(floatingHeader).toBeInTheDocument();
  });

  it("renders the unauthenticated (user) floating header component", async () => {
    await act(async () => {
      render(<FloatingHeader currentUser={mockAuthUser} />);
    });
    const floatingHeader = screen.getByTestId("floating-header");
    expect(floatingHeader).toBeInTheDocument();
  });

  it("hides (top:-49px) floating header component when scroll down viewport beyond threshold and bring it back if scroll it up (top:0px)", async () => {
    await act(async () =>
      render(<FloatingHeader currentUser={mockAuthUser} />)
    );
    const floatingHeader = screen.getByTestId("floating-header");

    expect(floatingHeader).toHaveStyle("top: 0px");

    // Scrolls downward (beyond threshold: 400 + 480 + 80 = 960)
    await act(async () => {
      Object.defineProperty(window, "scrollY", { value: 1000, writable: true });
      window.dispatchEvent(new Event("scroll"));
    });

    // Check if its hidden (moved -49px upwards viewport)
    await waitFor(() => {
      expect(floatingHeader).toHaveStyle("top: -49px");
    });

    // Scrolls upward
    await act(async () => {
      Object.defineProperty(window, "scrollY", { value: 500, writable: true });
      window.dispatchEvent(new Event("scroll"));
    });

    // Checks if header is back!
    await waitFor(() => {
      expect(floatingHeader).toHaveStyle("top: 0px");
    });
  });

  it("matches the authenticated (user) static header snapshot", async () => {
    await act(async () => {
      const { asFragment } = render(
        <FloatingHeader currentUser={mockUnauthUser} />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
