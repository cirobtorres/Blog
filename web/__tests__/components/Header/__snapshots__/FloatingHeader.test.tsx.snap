// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingHeader matches the authenticated (user) static header snapshot 1`] = `<DocumentFragment />`;
